### DISCUSSION TITLE
Making a get request fails if it is threaded

### DISCUSSION THREAD

**Discussion by enze-l** [micropython/micropython#10243] [Using MicroPython]
I want to make an get request to an api. It works as expected. But as soon as I make the request in a thread it fails.

Here is the code without threading:
```
import urequests
import _thread

def make_request():
    res = urequests.get("https://api.awattar.de/v1/marketdata")
    print(res.text)

make_request()

```
The code above works as expected. But now the threaded version:

```
import urequests
import _thread

def make_request():
    res = urequests.get("https://api.awattar.de/v1/marketdata")
    print(res.text)

_thread.start_new_thread(make_request, ())

```

This one throw the Exception: ```OSError: (-17040, 'MBEDTLS_ERR_RSA_PUBLIC_FAILED+MBEDTLS_ERR_MPI_ALLOC_FAILED')```

Am I doing something wrong or am I not aware of some behaviour? I would be greatfull for any help!

**Comment by jimmo** [comment]
@enze-l Sorry about the delay responding. This sounds like a bug. Which board/port are you using? Pico (W?), ESP32, STM32/pyboard, or something else?

**Reply by enze-l** [reply]
I am using a esp32 that is intigrated into this: https://www.shelly.cloud/de/products/product-overview/2xspl1pm/shelly-plus-1-pm . It contains an ESP32-U4WDH which is a single core variant of the esp32. I had to compile micropython by myself to be able to use it on this board because there aren't any precombiled firmware images as far as I know. I used this PR: https://github.com/micropython/micropython/pull/8365

**Comment by djmastertgv** [comment]
I also have this problem on my RPI Pico W, running Miropython 1.20
Same result when I call requests on thread, it's working fine if I call request without threading.

**Reply by scruss** [reply]
yours is a different issue (along with @GM-Script-Writer-62850): the OP is using a single core ESP32, not a dual core RP2

**Reply by GM-Script-Writer-62850** [reply]
why would you use thread on a single core chip? i assumed that OP's CPU supported SMT since it is a single core and _thread was used

**Reply by djmastertgv** [reply]
RP Pico W have dual core cpu, so I can use a second thread.

**Reply by scruss** [reply]
Not if you want to use wifi, though.

Best to ask your own question in the RP2040 section. This discussion is about ESP32

**Comment by GM-Script-Writer-62850** [comment]
i have been trying to work around this and wrote 2 theoretical workarounds
* https://github.com/orgs/micropython/discussions/11999

the idea is to use core 1 for the bulk of your work and offload stuff to core 0 while doing this i found a way to [break wifi](https://github.com/orgs/micropython/discussions/12023), maybe you a will have better luck

using a w5500 NIC i am able to use urequest on core 1 and it works, however it has been a stability nightmare with seemingly random hardlocks, if you do not need interrupts at all maybe it is stable, i think as soon as you use a single one ever in the script it can crash at any time after

* https://github.com/micropython/micropython/issues/10448
* https://github.com/micropython/micropython/issues/11974
